frontend/src/container/ForgotPasswordNoEmailNotif.jsx

```diff
@@ -9,94 +9,79 @@ import { resetBreadcrumbs, setConfig, setHeadTitle } from '../action-creator.syn
 import { CUSTOM_EVENT, IconButton, PAGE } from 'tracim_frontend_lib'
 import { getConfig } from '../action-creator.async'
 
-export class ForgotPasswordNoEmailNotif extends React.Component {
-  constructor (props) {
-    super(props)
+export const ForgotPasswordNoEmailNotif = props => {
+  useEffect(() => {
+    document.addEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, customEventReducer)
+    return () => {
+      document.removeEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, customEventReducer)
+    }
+  }, [])
 
-    document.addEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, this.customEventReducer)
-  }
+  useEffect(() => {
+    props.dispatch(resetBreadcrumbs())
+    if (!props.system.config.instance_name) loadConfig()
+  }, [])
 
-  customEventReducer = ({ detail: { type } }) => {
+  useEffect(() => {
+    updateHeadTitle()
+  }, [props.system.config.instance_name])
```

Comment:
seems good to me

will do

Reply:
Actually, it broke tests.
Indeed, the call to `updateHeadTitle` is now before it was initialized.
The quick fix would be declare updateHeadTitle right before the useEffect.
But I like having all the useEffect at the beginning of the component, it makes it easier to see all the life cycle methods of the component. Not sure if it's that important

Any advice @grignards?

Reply:
I do not have a very strong opinion, only that hooks are meant to group code by functionality in my mind which would lead to have the component code grouped by functionality rather than by "type".
e.g. in my mind components could be written like this (disclaimer: this is pseudo-code):
```javascript
const MyComponent = props => {
  // First functionality
  const [hello, setHello] = useState()
  const doIt = () => {
    // code here needing the state for example
  }
  useEffect(doIt, [])

  // Second functionality
  // …

  return (
    <div />
  )
}
```
The advantage is that as functionalities are grouped it should be easier to identify "patterns" and move them to custom hooks. Having custom hooks is useful in two cases I think:
 - when it helps to keep the code readable (to avoid having the main component function to be too long)
 - to factorize identical behavior between components.

@gbonaspetti @knodelle @Haskaris Do you have an opinion about this question of code structure?

Reply:
I don't have a strong opinion either, it seems nice to group by functionality but I think that many times the functionalities can mix together (or depend on each other) and in the end we will have all useEffect more or less at the end of the component

Reply:
I like the idea to be able "to identify "patterns" and move them to custom hooks"

Reply:
I grouped by functionalities instead of type